### ISSUE TITLE
Giving caseOrder error erroneously

### ISSUE THREAD

**Issue by echarrod** [go-critic/go-critic#1003]
Hello, I believe I am receiving a `caseOrder` warning erroneously.

I have the following switch statement:
```go
switch err := err.(type) {
	case nil:
		t.Fatalf("wanted %v, got %v", "a StrategyError", err)
	case *errorcodes.EditingError:
		if err.Code != errorcodes.ErrNoStrategy {
			t.Fatalf("wanted %v, got %v", "a NOT_SUPPORTED error", err)
		}
		t.Logf("All Good, this test did what it was supposed received an error of expected type %v", err)
	default:
		// unknown error
		t.Fatalf("wanted %v, got %v", "a NOT_SUPPORTED error", err)
	}
```

And go critic doesn't like it, saying `./internal/mod/t_errors_test.go:84:2: caseOrder: type is not defined nil`

If I switch the case statement with the `case *errorcodes.EditingError:` case, it complains about that one too: `caseOrder: type is not defined *errorcodes.EditingError`

From the example [here](https://go-critic.github.io/overview.html#caseOrder-ref), but I'm not sure how this relates to the code I'm looking at.

Also from the [positive tests](https://github.com/go-critic/go-critic/blob/master/checkers/testdata/caseOrder/positive_tests.go), I can't see why it is warning for my code.

**Comment by quasilyte** [comment]
It looks like there is an error during the source code loading (most likely, the typechecking).
Maybe Go loader can't find the dependencies for the `errorcodes` package.

I failed to reproduce it in isolation.
Do you have a standalone reproducer?

**Comment by cristaloleg** [comment]
Still an issue?

**Comment by echarrod** [comment]
Looks to be fine now, will close!
